integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-spark-service-bundle/nifi-spark-service/src/main/java/com/thinkbiganalytics/nifi/v2/spark/SparkJobserverService.java

```diff
@@ -0,0 +1,357 @@
+                        params.put(ISparkJobServerClientConstants.PARAM_TIMEOUT, syncTimeout);
+                    }
+
+                    getLogger().info(
+                        String.format("Creating %s %s with %s executors, %s cores and %s memory per executor on Spark Jobserver %s", contextType, contextName, numExecutors, numCPUCores, memPerNode,
+                                      jobServerUrl));
+                    contextRunning = client.createContext(contextName, params);
+                    getLogger().info(String.format("Created %s %s on Spark Jobserver %s", contextType, contextName, jobServerUrl));
+
+                } catch (Exception ex) {
+                    getLogger().error(ex.getMessage());
+                } finally {
+                    synchronized (sparkContextsStates) {
+                        if (contextRunning) {
+                            sparkContextsStates.get(contextName).isRunning = true;
+                            sparkContextsStates.get(contextName).setTimeoutSeconds(contextTimeout);
+                        } else {
+                            sparkContextsStates.remove(contextName);
+                        }
+                        sparkContextsStates.notifyAll();
+                    }
+                }
+            }
+            return contextRunning;
+        } else {
+            return true;
+        }
+    }
+
+    @Override
+    public boolean deleteContext(String contextName) {
+
+        Boolean contextDeleted = !checkIfContextExists(contextName);
+
+        if (!contextDeleted) {
+            try {
+                synchronized (sparkContextsStates) {
+                    contextDeleted = client.deleteContext(contextName);
+                    if (contextDeleted){
+                        sparkContextsStates.remove(contextName);
+                        getLogger().info(String.format("Deleted context %s from Spark Jobserver %s", contextName, jobServerUrl));
+                    }
+                }
+            } catch (Exception ex) {
+                getLogger().error(ex.getMessage());
+            }
+        }
+        return contextDeleted;
+    }
+
+    @Override
+    public boolean executeSparkContextJob(String appName, String classPath, String contextName, String args, boolean async) {
+
+        String id = contextName + System.nanoTime();
+
+        synchronized (sparkContextsStates) {
+            if (sparkContextsStates.containsKey(contextName)) {
+                sparkContextsStates.get(contextName).addExecutionLock(id);
+            }
+        }
+
+        SparkJobResult jobResult = null;
+        Boolean success = false;
+
+        try {
+            getLogger().info(String.format("Executing Spark App %s %s on context %s with args %s on Spark Jobserver %s", appName, classPath, contextName, args, jobServerUrl));
+
+            Map<String, String> params = new HashMap<String, String>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)